addons/AutoExportVersion/AutoExportVersion.gd

```diff
@@ -1,147 +1,185 @@
+####################################################################################################
+
+
+const _CURRENT_VERSION: String = "Current version: {version}"
+const _EMPTY_VERSION_ERROR: String = "Version string is empty.\nMake sure your 'get_version()' in '{script_path}' is configured properly."
+
+const _TOOL_MENU_ITEM_NAME: String = "AutoExport: Print Current Version"
+
+var _exporter: AutoExportVersionExporter
 
-var exporter: AEVExporter
 
 func _enter_tree() -> void:
-	exporter = AEVExporter.new()
-	exporter.plugin = self
-	add_export_plugin(exporter)
-	add_tool_menu_item("Print Current Version", self, "print_version")
-	
-	if not File.new().file_exists(VERSION_SCRIPT_PATH):
-		exporter.store_version(_fetch_version(PoolStringArray(), true, "", 0))
+	_exporter = AutoExportVersionExporter.new()
+	_exporter.plugin = self
+	add_export_plugin(_exporter)
+	add_tool_menu_item(_TOOL_MENU_ITEM_NAME, _tool_menu_print_version)
 
 func _exit_tree() -> void:
-	remove_export_plugin(exporter)
-	remove_tool_menu_item("Print Current Version")
-
-func print_version(ud):
-	var v = _fetch_version(PoolStringArray(), true, "", 0)
-	if v.empty():
-		OS.alert("Error fetching version. Check console for details.")
-	else:
-		OS.alert("Current game version: %s" % v)
-		print(v)
-
-class AEVExporter extends EditorExportPlugin:
-	var plugin
+	remove_export_plugin(_exporter)
+	remove_tool_menu_item(_TOOL_MENU_ITEM_NAME)
+
+func _tool_menu_print_version() -> void:
+	var version: String = get_version(PackedStringArray(), true, "", 0)
 	
-	func _export_begin(features: PoolStringArray, is_debug: bool, path: String, flags: int):
-		var version: String = plugin._fetch_version(features, is_debug, path, flags)
-		if version.empty():
-			push_error("Version string is empty. Make sure your _fetch_version() is configured properly.")
-		
-		store_version(version)
+	if version.is_empty():
+		printerr(_EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		OS.alert(_EMPTY_VERSION_ERROR.format({"script_path": get_script().get_path()}))
+		return
+	
+	printerr(_CURRENT_VERSION.format({"version":version}))
+	OS.alert(_CURRENT_VERSION.format({"version":version}))
+
 
-	func store_version(version: String):
-		var script = GDScript.new()
-		script.source_code = str("extends Reference\nconst VERSION = \"", version, "\"\n")
-		if ResourceSaver.save(VERSION_SCRIPT_PATH, script) != OK:
-			push_error("Failed to save version file. Make sure the path is valid.")
+class AutoExportVersionExporter extends EditorExportPlugin:
+	var plugin: EditorPlugin
+	
+	func _export_begin(features: PackedStringArray, is_debug: bool, path: String, flags: int) -> void:
+		if not plugin:
+			push_error("No plugin set in AutoExportVersionExporter")
+			return
+		
+		var version: String = plugin.get_version(features, is_debug, path, flags)
+		plugin.store_version(version, VersionStoreLocation.PROJECT_SETTING)
```

Comment:
Store location should be a variable/constant. If you put it at the bottom, under "unimportant plugin stuff", people will miss it...

Also make sure to document how to retrieve the version (i.e. example lines with `get_setting()` and script loading).
Also also the default should be `SCRIPT` IMO.

Reply:
>Store location should be a variable/constant. If you put it at the bottom, under "unimportant plugin stuff", people will miss it...

That is true. Did not realize it at all

>Also make sure to document how to retrieve the version (i.e. example lines with get_setting() and script loading).

I thought that would be better if it was documented in readme, but some example line would be great

>Also also the default should be SCRIPT IMO.

No problem with that

Reply:
Done in [this commit](https://github.com/Kubulambula/Godot-Auto-Export-Version/commit/fd5a12d0e6e94a80f14c742e8c90783d7d9e51df)